ddl/ddl_api.go

```diff
@@ -6400,6 +6455,129 @@ func (d *ddl) AlterTableAttributes(ctx sessionctx.Context, ident ast.Ident, spec
 	return errors.Trace(err)
 }
 
+func (d *ddl) AlterTableStatsOptions(ctx sessionctx.Context, ident ast.Ident, spec *ast.StatsOptionsSpec) error {
+	schema, tb, err := d.getSchemaAndTableByIdent(ctx, ident)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	meta := tb.Meta()
+	statsOptions := model.NewStatsOptions()
+	if !spec.Default {
+		if spec.StatsOptions == "" && meta.StatsOptions != nil {
```

Comment:
What's the case when the given spec is not the default and the options are empty?